features/steps/steps.py

```diff
@@ -283,4 +358,27 @@ def step_impl(context, representation_id):
                     errors.append(representation_shape_error(inst, representation_id))
     
     handle_errors(context, errors)
-    
+
+@then("It must have no duplicate points")
+def step_impl(context):
+    if getattr(context, 'applicable', True):
+        errors = []
+        for instance in context.instances:
+            entity_contexts = get_entity_contexts(context, instance)
+            precision = get_precision_from_contexts(entity_contexts)
+            points_coordinates = get_points(instance)
+            for i in itertools.combinations(points_coordinates, 2):
+                if math.dist(i[0], i[1]) < precision:
+                    errors.append(polyobject_structure_error(instance, points_coordinates, 'duplicate_points'))
```

Comment:
If we don't specify which points are the duplicates I think we should `break` here to not report a ton of duplicates on the same pline.

Reply:
Done.

Reply:
Done.